### DISCUSSION TITLE
Dynamic C# API Client Login question

### DISCUSSION THREAD

**Discussion by AndersChen123** [abpframework/abp#19339] [Q&A] [answered]
I am using dynamic api client in a WPF application, and it can login automatically after configuring the user information in appsettings.json, this is the content of appsettings.json
```
{
  "RemoteServices": {
    "Default": {
      "BaseUrl": "https://localhost:44320"
    }
  },
  "IdentityClients": {
    "Default": {
      "GrantType": "password",
      "ClientId": "BookStoreMVC_App",
      "ClientSecret": "1q2w3E*",
      "UserName": "admin",
      "UserPassword": "1q2w3E*",
      "Authority": "https://localhost:44320",
      "Scope": "BookStoreMVC"
    }
  }
}

```

But I want the user to enter the username and password manually, how can I achieve this?

Thanks.

**Comment by maliming** [comment, marked as answer]
hi

You can get an `access_token` by username and password.(password flow)
Then carry this token by overriding the `IIdentityModelAuthenticationService.`

**Reply by AndersChen123** [reply]
Thanks, that solves my problem.
